presto-orc/src/main/java/com/facebook/presto/orc/OrcRecordReader.java

```diff
@@ -257,12 +256,11 @@ public OrcRecordReader(
                 writeValidation);
 
         streamReaders = createStreamReaders(orcDataSource, types, hiveStorageTimeZone, presentColumnsAndTypes.build(), streamReadersSystemMemoryContext);
-        if (includedColumnHandles != null) {
-            for (int columnIdx = 0; columnIdx < streamReaders.length; columnIdx++) {
-                ColumnHandle columnHandle = includedColumnHandles.get(columnIdx);
-                if (columnHandle != null && streamReaders[columnIdx] != null && columnHandle.getReferencedSubfields() != null) {
-                    streamReaders[columnIdx].setReferencedSubfields(columnHandle.getReferencedSubfields(), 0);
-                }
+        for (Map.Entry<Integer, ColumnHandle> entry : includedColumnHandles.entrySet()) {
+            Integer columnIndex = entry.getKey();
+            ColumnHandle columnHandle = entry.getValue();
+            if (columnHandle.getReferencedSubfields() != null && streamReaders[columnIndex] != null) {
+                streamReaders[columnIndex].setReferencedSubfields(columnHandle.getReferencedSubfields(), 0);
```

Comment:
Still just minor commentary, but what I liked about the previous code was the for loop made it clear we were attempting to set every element of `streamReaders`.  With this code we are iterating over a `entrySet`, which is unordered, and results in unordered setting of elements of `streamReaders` array, which makes me wonder about the state of `streamReaders` after the loop, and if columnIndex can really be used to index into the `streamReaders` array.  With the for loop, I know I can index into this array.

Reply:
I wonder if this belongs in the `createStreamReaders` method, and we can put this in the single for-loop that exists in there.

Reply:
@tdcmeehan My reading of the code is that it was setting elements of the `streamReaders` array, but rather invoking `setReferencedSubfields` on some of the elements. More specifically, the elements that had a corresponding entry in `includedColumnHandles`. The original version was making a map lookup for every element of the `streamReaders` array and checking if the entry exists. The lookup and check are not necessary, because we can simply loop over `includedColumnHandles`. Does this make sense?

Reply:
@tdcmeehan 

> I wonder if this belongs in the `createStreamReaders` method, and we can put this in the single for-loop that exists in there.

This would make sense. I made this change. Would you take another look?